### PR TITLE
Introduce tip_header method in the HeaderService.

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

File: network-grpc/src/client.rs [hunk 1]
```diff
@@ -385,13 +385,19 @@ where
 
 impl<T, S, E> HeaderService<T> for Client<S, E>
 where
-    T: HasHeader,
```
Comment (mzabaluev) [inline network-grpc/src/client.rs]: Is this extra bound necessary?

File: network-grpc/src/client.rs [hunk 2]
```diff
@@ -385,13 +385,19 @@ where
...
+    T: Block + HasHeader,
```
Comment (NicolasDP) [inline network-grpc/src/client.rs]: this is not necessarily true (that a block has a header).

Why not parameterising the trait with a `H: Header` parameter ?

Comment (qnikst) [reply network-grpc/src/client.rs]: Fixed, it appeared to be a redundant change, left after previous work.

File: network-grpc/src/client.rs (hunk 1 above)
Comment (qnikst) [reply network-grpc/src/client.rs]: actually not, fixed

**Comment by qnikst** [comment]
Please take another look

**Pull request merged**